ops/model.py

```diff
@@ -985,6 +1001,55 @@ def location(self):
         return self._location
 
 
+class Container:
+    """Represents a named container in a unit.
+
+    Attributes:
+        name: The name of the container from metadata.yaml (eg, 'postgres').
+    """
+
+    def __init__(self, name, pebble_client=None):
+        self.name = name
+
+        if pebble_client is None:
+            socket_path = '/charm/containers/{}/pebble/.pebble.socket'.format(name)
+            pebble_client = pebble.PebbleClient(socket_path=socket_path)
+        self._pebble = pebble_client
+
+    @property
+    def pebble(self) -> 'pebble.PebbleClient':
+        """Return the low-level Pebble client instance for this container."""
+        return self._pebble
+
+    def autostart(self):
+        """Autostart all default services."""
+        self._pebble.autostart_services()
+
+    def start(self, *service_names: typing.List[str]):
+        """Start given service(s) by name."""
+        self._pebble.start_services(service_names)
+
+    def stop(self, *service_names: typing.List[str]):
+        """Stop given service(s) by name."""
+        self._pebble.stop_services(service_names)
+
+    def add_layer(self, setup: typing.Union[str, typing.Dict, 'pebble.Setup']):
+        """Dynamically add a service (Pebble) setup layer.
+
+        Args:
+            setup: A YAML string, setup dict, or pebble.Setup object
+                containing the Pebble setup layer to add.
```

Comment:
I now realize there's a small conceptual clash here.  In Pebble, the Setup type contains the entire Pebble setup, with all the layers and any additional global settings relevant to the entire process. Here, we're talking about a Layer, not a complete Setup.  I believe the Pebble code remains true to this idea, or is there an inconsistency on the other side as well?

Reply:
Good point -- a setup has a list of layers. I've changed to just `pebble.Layer` as that's all that we need here. And `get_setup` returns a flattened setup, which essentially means flattened to a single layer.